src/rust/engine/engine_pyo3/src/externs/interface/testutil.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::sync::Arc;
+
+use super::PyExecutor;
+use parking_lot::Mutex;
+use pyo3::exceptions::PyAssertionError;
+use pyo3::prelude::*;
+use pyo3::types::PyType;
+use testutil_mock::{StubCAS, StubCASBuilder};
+
+#[pyclass]
+pub struct PyStubCASBuilder {
+  builder: Arc<Mutex<Option<StubCASBuilder>>>,
+}
+
+#[pymethods]
+impl PyStubCASBuilder {
+  fn always_errors(&mut self) -> PyResult<PyStubCASBuilder> {
```

Comment:
It looks like the intention of this function might be to modify `self`, in which case it's possible to use `PyRef<Self>` as the receiver and return value:

```suggestion
  fn always_errors(this: PyRef<self>) -> PyResult<PyRef<Self>> {
```

... and just returning `this` at the end will return the same object instead of making a copy.

Reply:
... in fact, I think if you used `PyRefMut<Self>` then the `Arc<Mutex<...>>` may also not be necessary?

```rust
#[pyclass]
pub struct PyStubCASBuilder {
  builder: Option<StubCASBuilder>,
}

#[pymethods]
impl PyStubCASBuilder {
  fn always_errors(this: PyRefMut<Self>) -> PyRefMut<Self> {
    self.builder = self.builder.map(|builder| builder.always_errors());
    this
  }
```

Reply:
I tried this out but couldn't get it to compile. This is testutil code, so not a huge deal to have the Arc<Mutex>>. 

Thanks for the suggestion!